daemon/src/connection.rs

```diff
@@ -15,7 +15,7 @@ use xtra_productivity::xtra_productivity;
 
 struct ConnectedState {
     last_heartbeat: SystemTime,
-    _pulse_handle: RemoteHandle<()>,
+    tasks: Vec<RemoteHandle<()>>,
```

Comment:
What do you think of using the `Tasks` struct in more places and extend it with an API like:

```rust
impl Tasks {
	pub fn add(&mut self, f: impl Future) {
		let handle = f.spawn_with_handle();
		self.0.push(handle);
	}
}
```

Then we could do things like:

```rust
tasks.add(self.send_to_maker_ctx.attach(send_to_socket));
```

Which would be a bit shorter.

Reply:
great idea!

Reply:
Added the API, guarded by a `debug_assert!` to prevent calling `spawn_with_handle()` twice (as it makes no sense). Guard is covered with a test, and it was helpful when refactoring the code (I missed 2!)